api/v1beta1/machine_types.go

```diff
@@ -200,22 +200,70 @@ const (
+	// MachineNodeDeletedV1Beta2Reason surfaces when the node hosted on the machine has been deleted.
 	// Note: controllers can't identify if the Node was deleted by the controller itself, e.g.
 	// during the deletion workflow, or by a users.
 	MachineNodeDeletedV1Beta2Reason = ObjectDeletedV1Beta2Reason
 )
 
-// Machine's HealthCheckSucceeded and OwnerRemediated conditions and corresponding reasons that will be used in v1Beta2 API version.
-// Note: HealthCheckSucceeded and OwnerRemediated condition are set by the MachineHealthCheck controller.
+// Machine's HealthCheckSucceeded condition and corresponding reasons that will be used in v1Beta2 API version.
+// Note: HealthCheckSucceeded condition ia set by the MachineHealthCheck controller.
 const (
 	// MachineHealthCheckSucceededV1Beta2Condition is true if MHC instances targeting this machine report the Machine
 	// is healthy according to the definition of healthy present in the spec of the MachineHealthCheck object.
 	MachineHealthCheckSucceededV1Beta2Condition = "HealthCheckSucceeded"
 
+	// MachineHealthCheckPassedV1Beta2Reason surfaces when a machine passes all the health checks defined by a MachineHealthCheck object.
+	MachineHealthCheckPassedV1Beta2Reason = "HealthCheckPassed"
+
+	// MachineHealthCheckUnhealthyNodeV1Beta2Reason surfaces when the node hosted on the machine do not pass the health checks
+	// defined by a MachineHealthCheck object.
+	MachineHealthCheckUnhealthyNodeV1Beta2Reason = "UnhealthyNode"
+
+	// MachineHealthCheckNodeStartupTimeoutV1Beta2Reason surfaces when the node hosted on the machine does not appear within
+	// he timeout defined by a MachineHealthCheck object.
+	MachineHealthCheckNodeStartupTimeoutV1Beta2Reason = "NodeStartupTimeout"
+
+	// MachineHealthCheckNodeDeletedV1Beta2Reason surfaces when a MachineHealthCheck detect that the node hosted on the
+	// machine has been deleted while the Machine is still running.
+	MachineHealthCheckNodeDeletedV1Beta2Reason = "NodeDeleted"
+
+	// MachineHealthCheckManuallyRemediatedV1Beta2Reason surfaces a MachineHealthCheck detects a machine manually remediated
+	// via the remediate-machine annotation.
+	MachineHealthCheckManuallyRemediatedV1Beta2Reason = "ManuallyRemediated"
+)
+
+// Machine's OwnerRemediated conditions and corresponding reasons that will be used in v1Beta2 API version.
+// Note: OwnerRemediated condition is created by the MachineHealthCheck controller, and then Set by the Machine's owner controller.
+const (
 	// MachineOwnerRemediatedV1Beta2Condition is only present if MHC instances targeting this machine
 	// determine that the controller owning this machine should perform remediation.
 	MachineOwnerRemediatedV1Beta2Condition = "OwnerRemediated"
+
+	// MachineOwnerRemediatedWaitingForOwnerV1Beta2Reason surfaces the machine is waiting for the owner controller
+	// to start remediation.
+	MachineOwnerRemediatedWaitingForOwnerV1Beta2Reason = "WaitingForOwner"
```

Comment:
Would it be better to use "WaitingForRemediation". Currently we don't say what we're waiting for, we just say we're waiting for the owner

(I think the condition type already expresses that the owner is the one who is supposed to remediate, so we don't have to mention owner in the reason in my opinion)